modules/pricer/src/main/java/com/opengamma/strata/pricer/fx/SmileDeltaTermStructureParameters.java

```diff
@@ -5,479 +5,202 @@
-   * and {@code m} is the length of {@code delta}.
-   * <p>
-   * The default interpolator is used to interpolate in the time dimension. 
-   * 
-   * @param timeToExpiry  the time to expiry of each volatility smile
-   * @param delta  the delta at which the volatilities are given 
-   * @param volatility  the volatilities at each delta 
-   */
-  public SmileDeltaTermStructureParameters(double[] timeToExpiry, double[] delta, double[][] volatility) {
-    this(Long.toString(ATOMIC.getAndIncrement()), timeToExpiry, delta, volatility);
-  }
+public interface SmileDeltaTermStructureParameters {
 
   /**
-   * Constructor from name and market data.
-   * <p>
-   * The market date consists of time to expiry, delta and volatility.
-   * <p>
-   * The range of delta is common to all time to expiry.
-   * {@code volatility} should be {@code n * (2 * m + 1)}, where {@code n} is the length of {@code timeToExpiry}
-   * and {@code m} is the length of {@code delta}.
-   * <p>
-   * The default interpolator is used to interpolate in the time dimension. 
+   * Calculates the volatility at a given time/strike/forward from the term structure. 
    * 
-   * @param name  the name of smile parameter term structure 
-   * @param timeToExpiry  the time to expiry of each volatility smile
-   * @param delta  the delta at which the volatilities are given
-   * @param volatility  the volatilities at each delta 
+   * @param expiry  the time to expiry
+   * @param strike  the strike
+   * @param forward  the forward
+   * @return the volatility
    */
-  public SmileDeltaTermStructureParameters(String name, double[] timeToExpiry, double[] delta, double[][] volatility) {
-    ArgChecker.notNull(timeToExpiry, "time to expiry");
-    ArgChecker.notNull(delta, "delta");
-    ArgChecker.notNull(volatility, "volatility");
-    _name = ArgChecker.notNull(name, "name");
-    _timeToExpiry = ArgChecker.notNull(timeToExpiry, "timeToExpiry");
-    int nbExp = timeToExpiry.length;
-    ArgChecker.isTrue(volatility.length == nbExp,
-        "Volatility array length {} should be equal to the number of expiries {}", volatility.length, nbExp);
-    ArgChecker.isTrue(volatility[0].length == 2 * delta.length + 1,
-        "Volatility array {} should be equal to (2 * number of deltas) + 1, have {}",
-        volatility[0].length, 2 * delta.length + 1);
-    _volatilityTerm = new SmileDeltaParameters[nbExp];
-    for (int loopexp = 0; loopexp < nbExp; loopexp++) {
-      _volatilityTerm[loopexp] = SmileDeltaParameters.of(
-          timeToExpiry[loopexp], DoubleArray.copyOf(delta), DoubleArray.copyOf(volatility[loopexp]));
-    }
-    _timeInterpolator = DEFAULT_INTERPOLATOR_EXPIRY;
-    ArgChecker.isTrue(_volatilityTerm[0].getVolatility().size() > 1,
-        "Need more than one volatility value to perform interpolation");
-  }
+  public double getVolatility(double expiry, double strike, double forward);
```

Comment:
Elsewhere, we use `volatility` not `getVolatility`, because it involves computation. Same for `volatilityAndSensitivities` and the two smile methods